src/main/java/org/elasticsearch/percolator/PercolatorService.java

```diff
@@ -843,15 +847,29 @@ private InternalAggregations reduceAggregations(List<PercolateShardResponse> sha
             return null;
         }
 
+        InternalAggregations aggregations;
         if (shardResults.size() == 1) {
-            return shardResults.get(0).aggregations();
-        }
-
+            aggregations = shardResults.get(0).aggregations();
+        } else {
```

Comment:
I don't think reducers should affect settings in other aggregations (although its fine to validate settings in other aggregations). As a user I would be very confused if I set something and then the result wasn't what I expected because a reducer had slightly changed some settings. I do, however, think we should change the default of `min_doc_count` to be zero, just for all histograms, not just when used with certain reducers.

This should remove your concern here?